modules/apps/callbacks/types/events.go

```diff
@@ -0,0 +1,97 @@
+	//   "acknowledgement": the callback is executed on the acknowledgement of the packet
+	//   "timeout": the callback is executed on the timeout of the packet
+	//   "recv_packet": the callback is executed on the reception of the packet
+	AttributeKeyCallbackTrigger = "callback_trigger"
+	// AttributeKeySourceCallbackAddress denotes the source callback contract address
+	AttributeKeySourceCallbackAddress = "src_callback_address"
+	// AttributeKeyDestCallbackAddress denotes the destination callback contract address
+	AttributeKeyDestCallbackAddress = "dest_callback_address"
+	// AttributeKeyCallbackResult denotes the callback result:
+	//   "success": the callback is successfully executed
+	//   "failure": the callback is failed to execute
+	AttributeKeyCallbackResult = "callback_result"
+	// AttributeKeyCallbackError denotes the callback error message
+	// if no error is returned, then this key will not be included in the event
+	AttributeKeyCallbackError = "callback_error"
+	// AttributeKeyCallbackGasLimit denotes the custom gas limit for the callback execution
+	// if custom gas limit is not in effect, then this key will not be included in the event
+	AttributeKeyCallbackGasLimit = "callback_gas_limit"
+	// AttributeKeyCallbackPortID denotes the port ID of the packet
+	AttributeKeyCallbackPortID = "callback_port"
+	// AttributeKeyCallbackChannelID denotes the channel ID of the packet
+	AttributeKeyCallbackChannelID = "callback_channel"
+	// AttributeKeyCallbackSequence denotes the sequence of the packet
+	AttributeKeyCallbackSequence = "callback_sequence"
+)
+
+// EmitSourceCallbackEvent emits an event for a source callback
+func EmitSourceCallbackEvent(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	callbackTrigger string,
+	callbackData CallbackData,
+	err error,
+) {
+	emitCallbackEvent(ctx, packet, EventTypeSourceCallback, callbackTrigger, callbackData, err)
+}
+
+// EmitDestinationCallbackEvent emits an event for a destination callback
+func EmitDestinationCallbackEvent(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	callbackTrigger string,
+	callbackData CallbackData,
+	err error,
+) {
+	emitCallbackEvent(ctx, packet, EventTypeDestinationCallback, callbackTrigger, callbackData, err)
+}
+
+// emitCallbackEvent emits an event for a callback
+func emitCallbackEvent(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	callbackType string,
+	callbackTrigger string,
+	callbackData CallbackData,
+	err error,
+) {
+	success := err == nil
+	attributes := []sdk.Attribute{
+		sdk.NewAttribute(sdk.AttributeKeyModule, ModuleName),
+		sdk.NewAttribute(AttributeKeyCallbackTrigger, callbackTrigger),
+		sdk.NewAttribute(AttributeKeySourceCallbackAddress, callbackData.SrcContractAddr),
+		sdk.NewAttribute(AttributeKeyDestCallbackAddress, callbackData.DestContractAddr),
+		sdk.NewAttribute(AttributeKeyCallbackGasLimit, fmt.Sprintf("%d", callbackData.GasLimit)),
+		sdk.NewAttribute(AttributeKeyCallbackResult, fmt.Sprintf("%t", success)),
```

Comment:
Since `success` is a boolean, `fmt.Sprintf("%t", success)` will print `true` or `false` and not `success` or `failure` as mentioned in the documentation above, right?

Reply:
done